appendix-basic65.tex

```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
+120 GOTO 100
+130 PRINT "DONE."
+\end{screenoutput}
+
+BASIC 65 knows the difference between assignment and equality based on
+context. Consider this line of code:
+
+\begin{screenoutput}
+A = B = 10
+\end{screenoutput}
+
+BASIC 65 expects a statement, and notices a variable name followed by the
+{\bf =} symbol. It concludes that this is a statement assigning a value
+to the number variable \screentext{A}. It then expects a number expression on
+the right-hand side of the assignment, and notices the {\bf =} symbol
+is an operator in that expression. It concludes that the operation is an
+equality test, and proceeds to evaluate the expression and assign the result.
+
+The operators {\bf NOT}, {\bf AND}, {\bf OR} and {\bf XOR} can be used either
+as logical operators or as boolean operators. A logical operator join two
+conditional expressions as operands and evaluates to the logical comparison of
+their truth values.
+
+\begin{screenoutput}
+IF X=99 OR Y<5 THEN 130
+
+IF Y>10 AND Y<20 THEN 150
+\end{screenoutput}
+
+A boolean operator accepts two number operands and performs a calculation on
+the bits of the binary values.
+
+\begin{screenoutput}
+A=17
+PRINT A AND 20
+\end{screenoutput}
+
+Unlike other cases where operators have different behaviors based on how they
+are used, BASIC 65 does not need to determine whether these operators are
+behaving as logical operators or boolean operators. Because ``true'' and
+``false'' are represented by carefully chosen numbers, the logical operators
+have the same behavior whether their operands are conditional expressions or
+numbers. A ``true'' conditional expression is the number $-1$, which
+internally is a binary number with all bits set. The logical expression ``true
+and false'' is equivalent to the binary boolean expression {\tt \%....0000 \&
+\%....1111}. In this case, the {\bf AND} operator evaluates to 0, which is
+``false.''
+
+Conditional expressions evaluating to numbers can be used in some clever
+programming tricks. Consider this example:
+
+\begin{screenoutput}
+A = A - (B > 7)
+\end{screenoutput}
+
+This statement will increment the value in the \screentext{A} by 1 if the
+value in \screentext{B} is greater than 7. Otherwise it leaves it unchanged.
+If the sub-expression \screentext{B > 7} is true, then it evaluates to -1.
+\screentext{A - (-1)} is equivalent to \screentext{A + 1}. If the
+sub-expression is false, then it evaluates to 0, and \screentext{A - 0} is
+equivalent to \screentext{A}.
+
+When multiple operators are used in a single expression, the order in which
+they are evaluated is specified by {\em precedence}. For example, in the
+statement \screentext{A * A - B * B}, both multiplications will be performed
+first, then the subtraction. As in algebra, you can use parentheses to change
+the order of execution. In the expression \screentext{A * (A - B) * B}, the
+subtraction is performed first.
+
+The complete set of operators and their order of predecence are summarized in
```

Comment:
1. typo: precedence

2. british spelling: summarised

Reply:
Both fixed. Thanks for reading so closely! 😅